p2p/host/basic/basic_host.go

```diff
@@ -343,37 +360,77 @@ func makeUpdatedAddrEvent(prev, current []ma.Multiaddr) *event.EvtLocalAddresses
 	return &evt
 }
 
+func (h *BasicHost) makeSignedPeerRecord(evt *event.EvtLocalAddressesUpdated) (*record.Envelope, error) {
+	current := make([]multiaddr.Multiaddr, len(evt.Current))
+	for _, a := range evt.Current {
+		current = append(current, a.Address)
+	}
+
+	addrs := make([]multiaddr.Multiaddr, 0, len(current))
+	for _, a := range current {
+		if a == nil {
+			continue
+		}
+		addrs = append(addrs, a)
+	}
+
+	rec := peer.NewPeerRecord()
+	rec.PeerID = h.ID()
+	rec.Addrs = addrs
```

Comment:
This is a very unusual API. @yusefnapora what was the motivation for it? Why not `NewPeerRecord(id, addrs)`? Or just `&PeerRecord{ID: ..., Addrs: ..., Seq: TimestampSeq()}`?

Reply:
@Stebalien There is a `PeerRecordFromAddrInfo(peer.AddrInfi)` API which we weren't using before. Have fixed this.